ddtrace/contrib/aiohttp/patch.py

```diff
@@ -1,54 +1,127 @@
+import os
+
+from yarl import URL
+
 from ddtrace import config
 from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils.formats import asbool
 from ddtrace.vendor import wrapt
 
-from ...internal.utils.wrappers import unwrap
+from ...ext import SpanTypes
 from ...pin import Pin
+from ...propagation.http import HTTPPropagator
+from ..async_trace_utils import with_traced_module
+from ..trace_utils import ext_service
+from ..trace_utils import set_http_meta
+from ..trace_utils import unwrap
+from ..trace_utils import wrap
 
 
 log = get_logger(__name__)
 
-try:
-    # instrument external packages only if they're available
-    import aiohttp_jinja2
+propagator = HTTPPropagator()
 
-    from .template import _trace_render_template
-
-    template_module = True
-except ImportError:
-    template_module = False
-except Exception:
-    log.warning("aiohttp_jinja2 could not be imported and will not be instrumented.", exc_info=True)
-    template_module = False
 
 config._add(
     "aiohttp",
     dict(
-        distributed_tracing=True,
+        distributed_tracing_enabled=asbool(os.getenv("DD_AIOHTTP_DISTRIBUTED_TRACING", False)),
     ),
 )
 
 
+class _WrappedConnectorClass(wrapt.ObjectProxy):
+    def __init__(self, obj, pin):
+        super().__init__(obj)
+        pin.onto(self)
+
+    async def connect(self, req, *args, **kwargs):
+        pin = Pin.get_from(self)
+        with pin.tracer.trace("%s.connect" % self.__class__.__name__):
+            result = await self.__wrapped__.__class__.connect(self, req, *args, **kwargs)
+            return result
+
+    async def _create_connection(self, req, *args, **kwargs):
+        pin = Pin.get_from(self)
+        with pin.tracer.trace("%s._create_connection" % self.__class__.__name__):
+            result = await self.__wrapped__._create_connection(req, *args, **kwargs)
+            return result
+
+
+@with_traced_module
+async def _traced_clientsession_request(aiohttp, pin, func, instance, args, kwargs):
+    method: str = get_argument_value(args, kwargs, 0, "method")
+    url = URL(get_argument_value(args, kwargs, 1, "url"))  # url can be a str or yarl.URL
+    headers = kwargs.get("headers") or {}
+
+    # Create a new span and attach to this instance (so we can
+    # retrieve/update/close later on the response)
+    # Note that we aren't tracing redirects
+    with pin.tracer.trace(
+        "aiohttp.request", span_type=SpanTypes.HTTP, service=ext_service(pin, config.aiohttp)
+    ) as span:
+        if pin._config["distributed_tracing_enabled"]:
+            propagator.inject(span.context, headers)
+        set_http_meta(
+            span,
+            config.aiohttp,
+            method=method,
+            url=str(url),
+            query=url.query,
+            request_headers=headers,
+        )
+
+        resp = await func(*args, **kwargs)  # ClientResponse
```

Comment:
do we need to add headers back into `kwargs`? if they aren't in kwargs, then we create a new dict, and then it won't get used ?

Reply:
`inject` mutates the value passed to it but correct, in the case when headers are not supplied then this is a bug. Good catch!